monai/networks/trt_wrapper.py

```diff
@@ -0,0 +1,524 @@
+            # Use temporary directory for easy cleanup in case of external weights
+            with tempfile.TemporaryDirectory() as tmpdir:
+                onnx_path = Path(tmpdir) / 'model.onnx'
+                self.logger.info(f"Exporting to {onnx_path}, export args: {export_args}")
+                convert_to_onnx(
+                    model,
+                    input_example,
+                    filename=str(onnx_path),
+                    input_names=self.input_names,
+                    output_names=self.output_names,
+                    dynamo=dynamo,
+                    **export_args,
+                )
+                self.logger.info("Export to ONNX successful.")
+                engine_bytes = self._onnx_to_trt(str(onnx_path))
+
+        open(self.engine_path, 'wb').write(engine_bytes)
+
+
+def trt_forward(self, *argv, **kwargs):
+    """
+    Patch function to replace original model's forward() with.
+    Redirects to TrtWrappper.forward()
+    """
+    return self._trt_wrapper.forward(self, argv, kwargs)
+
+
+def trt_wrap(model, path, args=None, submodule=None, logger=None):
+    """
+    Instruments model or submodule with TrtWrappper and reppaces forward() with a hook.
+    Args:
+      model, path: passed to TrtWrappper().
+      args: dict : unpacked and passed to TrtWrappper().
+      submodule : Hierarchical id of submodule to convert, e.g. 'image_decoder.decoder'
+                  If None, TrtWrappper is applied to the whole model and returned.
+                  Otherwise, submodule is replaced in-place with TrtWrappper.
+    Returns:
+      Always returns same model passed in as argument. This is for ease of use in configs.
+    """
+    if args is None:
+        args = {}
+    orig_model = model
+    if trt_imported and polygraphy_imported and torch.cuda.is_available():
+        # if "path" filename point to existing file (e.g. checkpoint)
+        # it's also treated as dependency
+        if os.path.exists(path):
+            timestamp = os.path.getmtime(path)
+            if 'timestamp' in args:
+                timestamp = max(args['timestamp'], timestamp)
+            args['timestamp'] = timestamp
+
+        if submodule:
+            def find_sub(parent, submodule):
+                idx = submodule.find(".")
+                # if there is "." in name, call recursively
+                if idx != -1:
+                    parent_name = submodule[:idx]
+                    parent = getattr(parent, parent_name)
+                    submodule = submodule[idx + 1 :]
+                    return find_sub(parent, submodule)
+                return parent, submodule
+
+            path = path + '.' + submodule
+            parent, submodule = find_sub(model, submodule)
+            model = getattr(parent, submodule)
+
+        wrapper = TrtWrappper(model, path, logger=logger, **args)
+        model._trt_wrapper = wrapper
+        model.forward = MethodType(trt_forward, model)
+    return orig_model
```

Comment:
This return value looks a little bit weird to me, I would suggest to avoid `in place` modification for the input `model`, and return the wrapper model.
What do you think?
CC @ericspod 

Thanks.

Reply:
Well, my first take was, indeed, to wrap the model into TRTWrapper and return that. But then I also had to modify the whole model, replacing the wrapped model with the wrapper. That solution also had issues : resulting model would have its structure modified by wrapper insertion, so 1) checkpoint load could not be loaded after the wrapping, only before  and 2) if the pipeline utilized any other methods of the wrapped model, that would not be available via the wrapper. So I decided to patch the original model instead - much more flexible, 'pythonic' solution. 
Maybe trt_wrap function should  be renamed to trt_patch for clarity ?
Re return value: many in-place modification functions do return the argument back in Pytorch. My first take on trt_wrap() with model patch actually returned none - but while using it in config files I discovered it would not work unless I assign the value.

Reply:
OK, I see your point, thanks for the explanation.
`trt_patch()` looks better to me, we have some similar naming case in MONAI:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/bundle/workflows.py#L490

Thanks.

Reply:
@Nic-Ma : before I do this change, let's consider alternatives: 
1. trt_compile
2. trt_export

Reply:
The `trt_export` has already been used [here](https://github.com/Project-MONAI/MONAI/blob/de2a819e82e9c0575a959170d8e534fefe002d08/monai/bundle/scripts.py#L1559). Therefore, I would suggest to use the `trt_compile` or we need to delete this existing API.

Thanks 
Bin

Reply:
@binliunls : sure, let's go with trt_compile.